checkov/sca_package/output.py

```diff
@@ -57,10 +58,15 @@ def create_report_record(
     vulnerability_details: dict[str, Any],
     licenses: str,
     runner_filter: RunnerFilter | None = None,
+    image_details: ImageDetails | None = ImageDetails
 ) -> Record:
     runner_filter = runner_filter or RunnerFilter()
     package_name = vulnerability_details["packageName"]
     package_version = vulnerability_details["packageVersion"]
+    if hasattr(image_details, 'package_types'):
+        package_type = image_details.package_types.get(f'{package_name}@{package_version}', '')
+    else:
+        package_type = ''
```

Comment:
```suggestion
    if image_details:
        package_type = image_details.package_types.get(f'{package_name}@{package_version}', '')
    else:
        package_type = ''
```
probably that's what you wanted to do? because it can be either `None` or it will `ImageDetails`, which has the attribute `package_types`.

Reply:
Yeah that's what you'd expect, because `ImageDetails.package_types` has the `default_factory=dict`. 
But It just doesn't work! by default, it doesn't have a `package_types` attribute.. any idea why?

Reply:
is a test failing or mypy complains, or is there a problem during runtime?